contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,153 @@
+        pool.increaseCustodyAllowance(address(mplRewards), scaledDepositAmount);
+        mplRewards.stake(scaledDepositAmount);
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice function to start the cooldown process to withdraw
+    /// 1. lp lockup on deposit --> 90 days locked up and can't withdraw
+    /// 2. cool down period, call intend to withdraw -->
+    ///   must wait 10 days before withdraw after calling intend to withdraw function
+    /// 3. after cool down and past the lockup period,
+    ///    have 2 days to withdraw before cool down period restarts.
+    function signalIntentToWithdraw() external onlyPCVController {
+        pool.intendToWithdraw();
+    }
+
+    /// @notice function to cancel a withdraw
+    /// should only be used to allow a transfer when doing a withdrawERC20 call
+    function cancelWithdraw() external onlyPCVController {
+        pool.cancelWithdraw();
+    }
+
+    /// @notice withdraw PCV from Maple, only callable by PCV controller
+    /// @param to destination after funds are withdrawn from venue
+    /// @param amount of PCV to withdraw from the venue
+    function withdraw(address to, uint256 amount)
+        external
+        override
+        onlyPCVController
+    {
+        uint256 scaledWithdrawAmount = amount * scalingFactor;
+
+        mplRewards.getReward(); /// get MPL rewards
+        /// this call will withdraw amount of principal requested, and then send
+        /// over any accrued interest.
+        /// expected behavior is that this contract
+        /// receives either amount of USDC, or amount of USDC + interest accrued
+        /// if lending losses were taken, receive less than amount
+        mplRewards.withdraw(scaledWithdrawAmount); /// decreases allowance
+
+        /// withdraw from the pool
+        pool.withdraw(amount);
+        token.safeTransfer(to, amount);
+
+        emit Withdrawal(msg.sender, to, amount);
+    }
+
+    /// @notice withdraw all PCV from Maple
+    /// @param to destination after funds are withdrawn from venue
+    function withdrawAll(address to) external onlyPCVController {
+        uint256 amount = balance();
+        mplRewards.exit(); /// unstakes from Maple reward contract and claims rewards
+        /// this call will withdraw all principal,
+        /// then send over any accrued interest.
+        /// expected behavior is that this contract
+        /// receives balance amount of USDC, or amount of USDC + interest accrued
+        /// if lending losses were taken, receive less than amount
+        pool.withdraw(amount); /// call pool and withdraw entire balance
+
+        uint256 tokenBalance = IERC20(token).balanceOf(address(this));
+        token.safeTransfer(to, tokenBalance);
+
+        emit Withdrawal(msg.sender, to, tokenBalance);
+    }
+
+    /// permissionless function to harvest rewards before withdraw
+    function harvest() external {
+        mplRewards.getReward();
+
+        emit Harvest();
+    }
```

Comment:
add function here that allows governor to call in and make any arbitrary call with arbitrary calldata